golem/managers/proposal/plugins/scoring/pricings.py

```diff
@@ -29,23 +30,27 @@ def __str__(self) -> str:
         )
 
     def _calculate_cost(self, coeffs: LinearCoeffs) -> float:
-        average_duration_sec = self._average_duration.total_seconds()
+        average_duration_sec = Decimal(self._average_duration.total_seconds())
 
         average_initial_price = coeffs.price_initial
         average_duration_cost = coeffs.price_duration_sec * average_duration_sec
-        average_cpu_cost = coeffs.price_cpu_sec * self._average_cpu_load * average_duration_sec
+        average_cpu_cost = (
+            coeffs.price_cpu_sec * Decimal(self._average_cpu_load) * average_duration_sec
+        )
 
-        return average_initial_price + average_duration_cost + average_cpu_cost
+        return float(average_initial_price + average_duration_cost + average_cpu_cost)
```

Comment:
`LinearCoeffs` is a Dataclass that now uses Decimal now as it is more precise type and was helpful for PayDocuments validation.

`LinearAverageCostPricing` is a logic object that is used by other entities (e.g. scorers).

I decided to keep `LinearAverageCostPricing` interface as is to avoid spreading float->decimal changes to too many modules, while adjusting internal logic to account for change in `LinearCoeffs`.